src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKey.StandardProperties.cs

```diff
@@ -192,6 +192,24 @@ public int KeySize
                     throw errorCode.ToCryptographicException();
                 }
 
+                // The platform crypto provider always returns "0" for EC keys when asked for a key size. This
+                // has been observed in Windows 10 and most recently observed in Windows 11 22H2.
+                // The Algorithm NCrypt Property only returns the Algorithm Group, so that doesn't work either.
+                // What does work is the ECCCurveName.
+                if (keySize == 0 && Provider == CngProvider.MicrosoftPlatformCryptoProvider &&
+                    (AlgorithmGroup == CngAlgorithmGroup.ECDiffieHellman || AlgorithmGroup == CngAlgorithmGroup.ECDsa))
+                {
+
+                    string? curve = _keyHandle.GetPropertyAsString(KeyPropertyName.ECCCurveName, CngPropertyOptions.None);
+
+                    switch (curve)
+                    {
+                        case nameof(ECCurve.NamedCurves.nistP256): return 256;
+                        case nameof(ECCurve.NamedCurves.nistP384): return 384;
+                        case nameof(ECCurve.NamedCurves.nistP521): return 521;
+                    }
```

Comment:
What if the curve isn't one of these? We'll still return 0... that's ok?

Reply:
The TPM 2.0 spec defines a few elliptic curves. The NIST curves I've defined here are the only named curves that we have overlapped named curve support for. Worst case, that's the current behavior of the property now, so we're not in any worse position. If customers identify other curves that CNG and TPMs that we can meaningfully support, we can add them in the future.

Reply:
I believe the built-in KSP supports all 5 NIST curves (these three, plus 192 and 224) and nothing else.  So we might want to add 192/224 for completion's sake.

And I think Win8 didn't have (good?) support for EC keys in the TPM.

Reply:
> So we might want to add 192/224 for completion's sake.

Seems reasonable but I did not add tests for these for similar reasons I did not add a test for P521. Doesn't Work On My Machine.